docs/user-guide/sync-options.md

```diff
@@ -214,3 +214,17 @@ spec:
 ```
 
 The example above shows how an ArgoCD Application can be configured so it will ignore the `spec.replicas` field from the desired state (git) during the sync stage. This is achieve by calculating and pre-patching the desired state before applying it in the cluster. Note that the `RespectIgnoreDifferences` sync option is only effective when the resource is already created in the cluster. If the Application is being created and no live state exists, the desired state is applied as-is.
+
+## Create Namespace
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  namespace: test
+spec:
+  syncPolicy:
+    syncOptions:
+    - CreateNamespace=true
+```
+The example above shows how an ArgoCD Application can be configured so it will create a namespace called test, if it doesn't already exist and then make sure the application is created in that namespace. Without this, declared in a yaml or passed into the cli via `--sync-option CreateNamespace=true`, if the namespace does not exist, the application will fail to sync.
```

Comment:
Thx! for adding this info, very imp feature.
Can you rephrase this line for better clarity? 
```if it doesn't already exist and then make sure the application is created in that namespace```
to something like this
`if the namespace doesn't exist, then argocd will create a namespace declared in the YAML and deploy the application in the namespace. 
`

`
Without this, declared in a yaml or passed into the cli via `--sync-option CreateNamespace=true`, if the namespace does not exist, the application will fail to sync.
`
to something like this
`
You can add the sync option `CreateNamespace=true` in the YAML  or passed into the cli via --sync-option CreateNamespace=true`. 
`

Reply:
```suggestion
The example above shows how an Argo CD Application can be configured so it will create namespaces for the Application resources if the namespaces don't exist already. Without this either declared in the Application manifest or passed in the cli via `--sync-option CreateNamespace=true`, the Application will fail to sync if the resources' namespaces do not exist.
```

How about this? It captures the fact that `CreateNamespace=true` will create the namespace(s) for all resources (including but not limited to the `test` namespace, since resources can designate other namespaces as their destinations).